hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManagerServiceGrpc.java

```diff
@@ -126,20 +127,19 @@ public void submitRequest(OMRequest request,
       omResponse =
           UserGroupInformation.getCurrentUser().doAs(
               (PrivilegedExceptionAction<OMResponse>) () -> {
-                try {
                   return this.omTranslator.
                       submitRequest(NULL_RPC_CONTROLLER, request);
-                } catch (Throwable se) {
-                  Throwable e = se.getCause();
-                  if (e == null) {
-                    throw new IOException(se);
-                  } else {
-                    throw e instanceof IOException ?
-                        (IOException) e : new IOException(se);
-                  }
-                }
               });
     } catch (Throwable e) {
+      IOException ex = new IOException(e.getCause());
+      responseObserver.onError(Status
+          .INTERNAL
+          .withDescription(ex.getMessage())
+          .asRuntimeException());
+
+      // setting the omResponse, unused since we are generating
+      // standard model gRPC error through grpc Status and
```

Comment:
So ServiceException with LeaderNotReady/NotLeader will be thru Grpc errors .
And for others like business logic errors we donot throw, so i think we can remove setting the below line correct?

Reply:
Thanks @bharatviswa504  for your review and comments!  Right, for the expected exceptions for handling Service exceptions we handle with standard gRPC error `onError`.  For others, the error is returned through the OmResponse (status not ok with exception in message).  Yes, the lines setting the error for the omResponse is no longer needed (legacy code).  I'll cleanup removing lines 140 - 145 (setting the omResponse error) and the method `createErrorResponse` as well, as unneeded.